src/main/java/com/couchbase/lite/Manager.java

```diff
@@ -342,8 +400,16 @@ public boolean replaceDatabase(String databaseName, String databaseDir) {
             return false;
         }
 
-        if(!db.open()){
-            Log.w(Database.TAG, "Failed to open database");
+        boolean isOpen = false;
+        CouchbaseLiteException error = null;
+        try {
+            isOpen = db.open();
+        } catch (CouchbaseLiteException e) {
+            error = e;
```

Comment:
Because isOpen could be false without throwing an exception (I haven't fixed to throw an exception for all error cases).